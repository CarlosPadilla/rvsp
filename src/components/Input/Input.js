import React from 'react';
import PropTypes from 'prop-types';

const Input = props => (
  <div>
    <label>{props.label}</label>
    <input
      id={props.field}
      className={props.className}
      type={props.type}
      value={props.value}
      onChange={props.onChange}
    />
  </div>
);

Input.defaultProps = {
  value: '',
};

Input.propTypes = {
  field: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

export default Input;
