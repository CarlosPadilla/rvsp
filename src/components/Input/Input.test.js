import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';

describe('Input Component', () => {

  const value = ''
  const onInputChange = jest.fn()

  const app = shallow(
    <Input
      label="Nimi"
      field="name"
      className="input-name"
      type="text"
      value={value}
      onChange={onInputChange}
    />
  )

  it('renders', () => {
    expect(app).toMatchSnapshot()
  })

})
