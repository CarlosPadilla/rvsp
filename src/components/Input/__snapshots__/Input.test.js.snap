// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    className="input-name"
    field="name"
    label="Nimi"
    onChange={[Function]}
    type="text"
    value=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          Nimi
</label>,
        <input
          className="input-name"
          id="name"
          onChange={[Function]}
          type="text"
          value=""
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Nimi",
        },
        "ref": null,
        "rendered": "Nimi",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-name",
          "id": "name",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Nimi
</label>,
          <input
            className="input-name"
            id="name"
            onChange={[Function]}
            type="text"
            value=""
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Nimi",
          },
          "ref": null,
          "rendered": "Nimi",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-name",
            "id": "name",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
